common/test/acceptance/tests/discussion/test_cohort_management.py

```diff
@@ -294,6 +304,177 @@ def test_add_new_cohort(self):
             1
         )
 
+    def _verify_cohort_settings(
+            self,
+            cohort_name,
+            assignment_type,
+            new_cohort_name=None,
+            new_assignment_type=None,
+            verify_updated=False
+    ):
+
+        """
+        Create a new cohort and verify the new and existing settings.
+        """
+        start_time = datetime.now(UTC)
+        self.assertFalse(cohort_name in self.cohort_management_page.get_cohorts())
+        self.cohort_management_page.add_cohort(cohort_name, assignment_type=assignment_type)
+        # After adding the cohort, it should automatically be selected
+        EmptyPromise(
+            lambda: cohort_name == self.cohort_management_page.get_selected_cohort(), "Waiting for new cohort to appear"
+        ).fulfill()
+        self.assertEqual(0, self.cohort_management_page.get_selected_cohort_count())
+        # After adding the cohort, it should automatically be selected and its
+        # assignment_type should be "manual" as this is the default assignment type
+        msg = "Waiting for currently selected cohort assignment type"
+        EmptyPromise(
+            lambda: assignment_type == self.cohort_management_page.get_cohort_associated_assignment_type(), msg
+        ).fulfill()
+        # Go back to Manage Students Tab
+        self.cohort_management_page.select_manage_settings()
+        self.cohort_management_page.add_students_to_selected_cohort([self.instructor_name])
+        # Wait for the number of users in the cohort to change, indicating that the add operation is complete.
+        EmptyPromise(
+            lambda: 1 == self.cohort_management_page.get_selected_cohort_count(), 'Waiting for student to be added'
+        ).fulfill()
+        self.assertEqual(
+            self.event_collection.find({
+                "name": "edx.cohort.created",
+                "time": {"$gt": start_time},
+                "event.cohort_name": cohort_name,
+            }).count(),
+            1
+        )
+        self.assertEqual(
+            self.event_collection.find({
+                "name": "edx.cohort.creation_requested",
+                "time": {"$gt": start_time},
+                "event.cohort_name": cohort_name,
+            }).count(),
+            1
+        )
+
+        if verify_updated:
+            self.cohort_management_page.select_cohort(cohort_name)
+            self.cohort_management_page.select_cohort_settings()
+            self.cohort_management_page.set_cohort_name(new_cohort_name)
+            self.cohort_management_page.set_assignment_type(new_assignment_type)
+            self.cohort_management_page.save_cohort_settings()
+
+            # new cohort name should be present in the list of cohorts. But when we try to update an existing cohort
+            # with an empty cohort name then existing/old cohort name will be used because cohort name can't ba empty
+            new_cohort_name = new_cohort_name or cohort_name
```

Comment:
@explorerleslie was this a product decision to allow the save if the cohort name is empty (as opposed to showing a validation error), and just stick with the original name?